GHzDACs/ghz_fpga_server.py

```diff
@@ -1631,16 +1655,100 @@ def dac_run_sram(self, c, data, loop=False, blockDelay=0):
         """
         if not len(data):
             return
-        
-        if loop:
-            # make sure data is at least 20 words long by repeating it
-            data *= (20-1)/len(data) + 1
-        else:
-            # Set data at least 20 words long by repeating first value
+
+        if len(data) < 20:
+            # pad out to 20 with zeros
+            d2 = np.zeros(20)
+            d2[0:len(data)] = data
+            data = d2
             data = list(data) + [data[0]] * (20-len(data))
-        
+
         dev = self.selectedDAC(c)
         yield dev.runSram(data, loop, blockDelay)
+
+    @setting(2081, 'DAC Write SRAM', data='*w')
+    def dac_write_sram(self, c, data):
+        """ Write the given data to SRAM.
+
+        Note that if your board is currently running, it will immediately start
+        playing the new SRAM, wherever it is, so use with caution.
+        """
+        dev = self.selectedDAC(c)
+        yield dev._sendSRAM(np.array(data, dtype='<u4').tostring())
+
+    @setting(1082, "Add Jump Table Entry", name='s', arg=['ww{IDLE}', 'www{JUMP}', 'w{NOP,END}', 'wwww{CYCLE}'])
+    def add_jt_entry(self, c, name, arg=None):
+        """ Add a new jump table entry.
+
+        Entry types are: IDLE, JUMP, NOP, END, CYCLE, CHECK, RAMP
+        (as of now, CHECK and RAMP are unimplemented)
+
+        The arguments vary depending on the command, see the JT docs for
+        details. Note that from addresses and to addresses are to be specified
+        in nanoseconds (though still in words). They will be divided by 4 to
+        convert to SRAM addresses.
+        """
+        dev = self.selectedDAC(c)
+        assert(isinstance(dev, dac.DAC_Build13))
+        d = c.setdefault(dev, {})
+        entries = d.setdefault('jt_entries', [])
+        if name == 'CHECK':
+            raise NotImplementedError("Check not implemented yet")
+        elif name == 'RAMP':
+            raise NotImplementedError("Ramp not implemented yet")
+        elif name == 'IDLE':
+            from_address_ns, idle_duration_ns = arg
+            from_address = from_address_ns // 4
```

Comment:
Should we assert here that `from_address_ns` is a multiple of 4? Also below for other address parameters.